File: test-utils/src/main/java/ru/virgil/testutils/fluent_request/FinalStepNoType.java
package ru.virgil.testutils.fluent_request;

import com.google.common.truth.Truth;
import lombok.RequiredArgsConstructor;
import org.springframework.test.web.servlet.MvcResult;
import org.springframework.test.web.servlet.ResultMatcher;

@RequiredArgsConstructor
public class FinalStepNoType implements FinalStep<MvcResult> {

    private final RequestModel requestModel;
    private final Requester requester;

    public MvcResult expect(ResultMatcher... expectMatchers) throws Exception {
        Truth.assertThat(requestModel.getResponseJavaType()).isNull();
        requestModel.setResultMatchers(expectMatchers);
        return requester.makeRequest(requestModel);
    }
}

File: tools/src/main/kotlin/ru/virgil/spring_tools/tools/testing/fluent_request/FinalStepNoType.kt
package ru.virgil.spring_tools.tools.testing.fluent_request

import com.google.common.truth.Truth
import org.springframework.test.web.servlet.MvcResult
import org.springframework.test.web.servlet.ResultMatcher

class FinalStepNoType(
    private val requestModel: RequestModel,
    private val requester: Requester,
) : FinalStep<MvcResult> {

    @Throws(Exception::class)
    override fun expect(vararg expectMatchers: ResultMatcher): MvcResult {
        Truth.assertThat(requestModel.responseJavaType).isNull()
        requestModel.resultMatchers = (expectMatchers.toList())
        return requester.makeRequest(requestModel)
    }

}
